pkg/scheduler/partition.go

```diff
@@ -339,13 +339,18 @@ func (pc *PartitionContext) AddApplication(app *objects.Application) error {
 		return fmt.Errorf("failed to find queue %s for application %s", queueName, appID)
 	}
 
-	// set resources based on tags, but only if the queue is dynamic (unmanaged)
-	if queue.IsManaged() {
-		log.Log(log.SchedQueue).Warn("Trying to set resources on a queue that is not an unmanaged leaf",
-			zap.String("queueName", queue.QueuePath))
-	} else {
-		queue.SetResources(app.GetGuaranteedResource(), app.GetMaxResource())
+	guaranteedRes := app.GetGuaranteedResource()
+	maxRes := app.GetMaxResource()
+	if guaranteedRes != nil || maxRes != nil {
+		// set resources based on tags, but only if the queue is dynamic (unmanaged)
+		if queue.IsManaged() {
+			log.Log(log.SchedQueue).Warn("Trying to set resources on a queue that is not an unmanaged leaf",
+				zap.String("queueName", queue.QueuePath))
+		} else {
+			queue.SetResources(guaranteedRes, maxRes)
```

Comment:
This is unrelated to this PR, but what happens if this dynamic queue is `@recovery@` queue? IIRC, recovery queue has no quota

Reply:
Yes, you're right. Perhaps we should filter for the recovery queue. I think I'll create a separate JIRA for that.